kustomize/go.mod

```diff
@@ -9,47 +9,17 @@ require (
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.0
 	golang.org/x/text v0.3.7
-	sigs.k8s.io/kustomize/api v0.11.5
-	sigs.k8s.io/kustomize/cmd/config v0.10.7
-	sigs.k8s.io/kustomize/kyaml v0.13.7
```

Comment:
I don't recall these lines disappearing in your original PR, why do they get removed and can we keep them in?

Reply:
> can we keep them in?

I think it is not better.
Go package dependency in the same repo like `sigs.k8s.io/kustomize/kyaml` was kept to the newest code by Go modules `replace` until now. `replace` directive ignores package version in `go.mod`.
If we delete `replace` and keep the above lines in `go.mod` file, we need to update all dependency package versions included in this repo in *all* `go.mod` files when releasing.

Reply:
The tests may be passing, but these modules are all released independently and I'm concerned that removing it will have implications for the release process.

Besides those concerns, it is easier to reason about dependencies with these lines kept in, and I would really prefer not to remove them.

Reply:
> If we delete replace and keep the above lines in go.mod file, we need to update all dependency package versions included in this repo in all go.mod files when releasing.

Right now during the release process, we remove the replace statements and update all the dependencies anyways, so this is not so much a concern for me.

Reply:
@natasha41575 
I found old version dependencies in a few files. https://github.com/kubernetes-sigs/kustomize/pull/4692/files#diff-af0240957af36b1a68b96de432e777a7597f577f8a1113379f04a1121e01a84cL8
I updated these dependencies, but do you have any other opinions?

Reply:
I'm happy to have updated dependencies, thanks for checking!